src/dotty/tools/dotc/core/Names.scala

```diff
@@ -221,13 +221,15 @@ object Names {
   private var size = 1
 
   /** The hash of a name made of from characters cs[offset..offset+len-1].  */
-  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int =
-    if (len > 0)
-      (len * (41 * 41 * 41) +
-        cs(offset) * (41 * 41) +
-        cs(offset + len - 1) * 41 +
-        cs(offset + (len >> 1)))
-    else 0
+  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int = {
+    var i = offset
+    var hash = 0
+    while (i < len) {
```

Comment:
Original loop in the code you linked:
```scala
var i = 0
while (i < len) {
  h = ... cs(i + offset)
}
```
If you change `i` to start at `offset` then you also need to replace `len` by `len + offset`